src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -5,52 +5,87 @@
 
 import java.math.BigDecimal;
 import java.util.Currency;
+import java.util.List;
 import java.util.NoSuchElementException;
 
 import static org.junit.Assert.*;
 
-public class ArrayListProductDaoTest
-{
+public class ArrayListProductDaoTest {
     private ProductDao productDao;
+    Product product;
 
     @Before
     public void setup() {
         productDao = new ArrayListProductDao();
-    }
-    @Test (expected = NoSuchElementException.class)
-    public void testGetProducts(){
         Currency currency = Currency.getInstance("USD");
-        Product product = new Product("test", "HTC EVO Shift 4G", new BigDecimal(320), currency, 3, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/HTC/HTC%20EVO%20Shift%204G.jpg");
+        product = new Product("test", "HTC EVO Shift 4G", new BigDecimal(320), currency, 3, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/HTC/HTC%20EVO%20Shift%204G.jpg");
+    }
+
+    @Test
+    public void testGetExistingProducts() {
         productDao.save(product);
         assertEquals("test", productDao.getProduct(product.getId()).getCode());
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void testGetNonExistingProducts() {
+        productDao.save(product);
         productDao.getProduct(product.getId() + 1);
     }
+
+    @Test(expected = NoSuchElementException.class)
+    public void testGetProductNullId() {
+        productDao.getProduct(null);
+    }
+
     @Test
     public void testFindProductsHasResults() {
         assertFalse(productDao.findProducts().isEmpty());
     }
+
     @Test
-    public void testSaveExistingProduct(){
-        Currency currency = Currency.getInstance("USD");
-        Product product = new Product("test", "HTC EVO Shift 4G", new BigDecimal(320), currency, 3, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/HTC/HTC%20EVO%20Shift%204G.jpg");
+    public void testFindProductsReturnsNotNullPrice() {
+        List<Product> products = productDao.findProducts();
+        for (Product tempProduct : products) {
+            assertNotNull(tempProduct.getPrice());
+            assertTrue(tempProduct.getStock() > 0);
+        }
```

Comment:
You can use .allMatch() option of streams: assertTrue(products.stream().allMatch(product -> product.getPrice() != null && product.getStock() > 0)